crates/reporting/src/error/type.rs

```diff
@@ -675,9 +675,24 @@ fn to_expr_report<'b>(
             }
         }
         Expected::FromAnnotation(name, _arity, annotation_source, expected_type) => {
+            use roc_can::pattern::Pattern;
             use roc_types::types::AnnotationSource::*;
 
+            let is_suffixed = match &name.value {
+                Pattern::Identifier(symbol) => symbol.as_str(alloc.interns).starts_with("#!"),
```

Comment:
Similarly here - is there a way that we can get the info we need here, without inspecting a string like this?